frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-ref.ts

```diff
@@ -0,0 +1,77 @@
+import { K8sModel } from '../../api/common-types';
+import {
+  K8sResourceKindReference,
+  K8sGroupVersionKind,
+  K8sResourceCommon,
+} from '../../extensions/console-types';
+
+/**
+ * Provides a reference string that uniquely identifies the group, version, and kind of a k8s resource.
+ * @param group Pass group of k8s resource or model.
+ * @param version Pass version of k8s resource or model.
+ * @param kind Pass kind of k8s resource or model.
+ * @return The reference for any k8s resource i.e `group~version~kind`.
+ * * */
+export const getReference = (
+  group: string,
+  version: string,
+  kind: string,
+): K8sResourceKindReference => [group, version, kind].join('~');
```

Comment:
you already have types in `k8s-types.ts`

so it can be just 
```suggestion
export const getReference: GetReference = (
  group,
  version,
  kind,
) => [group, version, kind].join('~');
```

same for the rest

Reply:
Also I wonder do we actually need `getReference` utility ? I think we wanted to make the reference form `group~version~kind` an internal representation so we should not expose utility functions.

I know we already use the reference in exposed hooks, but how about updating those hooks ?
 
```js
export type UseK8sModel = (groupVersionKind?: GroupVersionKind | K8sGroupVersionKind) => [K8sModel, boolean]; //we can use reference (old format) or K8sGroupVersionKind
```

```js
export type WatchK8sResource = {
  kind: K8sResourceKindReference | K8sGroupVersionKind;  //we can use reference (old format) or K8sGroupVersionKind
  name?: string;
  namespace?: string;
  isList?: boolean;
  selector?: Selector;
  namespaced?: boolean;
  limit?: number;
  fieldSelector?: string;
  optional?: boolean;
};
```

@spadgett

Reply:
I agree `K8sGroupVersionKind` does make things easier for us as an exposed API as it does not rely on custom separator characters. Although I'm not sure we can drop `K8sResourceKindReference` in the SDK as it's a widely used mechanism today.

However, if we do reduce the means to which to craft it (ie drop/internalize `getReference*`) then we can migrate forward over time. Allowing older code to still provide the `A~B~C` format, but not make it easy to do? We might lose a bit of control over it if some crafty developers see it as an accepted format and start making it up themselves -- at least with the utility it's easy to base it and control it (sorta). 🤔

Reply:
we can clearly mark the old format as deprecated now and get rid of it in the future.
Am all in for making it harder to use the old format by not providing these utility functions.

Reply:
I'm good with this suggestion. We should work through the APIs and see where we expose references to add `K8sGroupVersionKind`. I know minimally we have it for `useK8sModel`, `useK8sWatchResource(s)`, and `ResourceLink` in exposed APIs.

Reply:
have made the changes as suggested https://github.com/openshift/console/pull/10303/commits/c9b96305bbe37b6ace0697c368a3ddbe88712d11.

PTAL when you get time , thanks :)